src/components/trade/Price.tsx

```diff
@@ -230,16 +230,21 @@ export const Price: React.FC<Props> = ({
     precision: DEFAULT_PRECISION,
   })
 
-  const hideInverse = (!isPriceInverted && wasPriorityAdjusted) || (isPriceInverted && !wasPriorityAdjusted)
-  const hideDirect = (isPriceInverted && wasPriorityAdjusted) || (!isPriceInverted && !wasPriorityAdjusted)
+  // Show Inverse Price ... IF
+  // DIRECT + ADJUSTED || INVERSE + UNADJUSTED
+  // aka
+  // 1. price = DIRECT && market = ADJUSTED
+  // 2. price = INVERSE && market = UNADJUSTED
+  const marketAdjustedisPriceInverted =
```

Comment:
I understand now, and is more clear. Thanks for reducing the duplication and comments
`marketAdjustedisPriceInverted`

My suggestion, to keep doing it more readable is to get rid everywhere of `wasPriorityAdjusted` with the only exception of where this data is actually calculated: In the trade widget index.ts
From the subcomponents point of view, they don't need this information and it really makes it less difficult to reason about.

The index.ts would need to do probably something else. Probably is just adjusting `priceIsInverse` to have a similar logic as the one you did in this line. 

Probably, something like:
```js
const priceIsInverted = priceShown === 'INVERSE'
const priorityWasAdjusted = baseToken == receiveToken // Note you didn't need it as part of the function

// price is inverted, either because it was inverted by the user, or the token priority made the base/quote selection to change
const priceIsInverse = priceIsInverted && !priorityWasAdjusted || !priceIsInverted && priorityWasAdjusted`
```

See how now the children don't need to know "the reason" why is inverted. Only that the price is the other way around.